### PR TITLE
ResourcePathProcessor folder cleanup

### PR DESCRIPTION
**Pull request by CyberiaResurrection**

### CODE REVIEW

File: src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php [hunk 1]
```diff
@@ -358,19 +361,30 @@ private static function getTypeAndValidateKeyValue($value, $tokenId, &$outValue,
...
         $keyString = $skipToken;
         return self::parseAndVerifyRawKeyPredicate($keyString, $isKey, $keyDescriptor);
     }
 
+    /**
+     * @param ExpressionLexer $expressionLexer
+     * @return \POData\UriProcessor\QueryProcessor\ExpressionParser\ExpressionToken
+     * @throws ODataException
+     */
+    private static function getNextLexerToken(ExpressionLexer $expressionLexer): ExpressionToken
```
Comment (c-harris) [inline src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php]: Shouldn't this be move to next not get next? As it permenatly moves the iterator?

File: src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php [hunk 2]
```diff
@@ -400,11 +414,12 @@ public function areNamedValues()
      *
      * @return int
      */
-    public function valueCount()
+    public function valueCount(): int
```
Comment (c-harris) [inline src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php:417]: Does this always return?

Comment (c-harris) [reply src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php:417]: Additionally if it is empty wouldn't count just return 0 anyway.

Comment (CyberiaResurrection) [reply src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php:417]: Take a look inside the guts of valueCount - namedValues takes precedence over positionalValues.

File: src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php (hunk 1 above)
Comment (CyberiaResurrection) [reply src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/KeyDescriptor.php]: Changed to toNextLexerToken.

**Pull request merged**